slack-base/src/main/java/com/hubspot/slack/client/models/interaction/SlackInteractiveCallback.java

```diff
@@ -11,18 +13,15 @@
 
 @JsonTypeInfo(use = Id.NAME, include = As.EXISTING_PROPERTY, property = "type", visible = true)
 @JsonSubTypes({
+    @Type(value = BlockActions.class, name = "block_actions"),
     @Type(value = InteractiveAction.class, name = "interactive_message"),
     @Type(value = DialogSubmission.class, name = "dialog_submission"),
     @Type(value = MessageAction.class, name = "message_action")
 })
 public interface SlackInteractiveCallback {
   InteractiveCallbackType getType();
-
-  String getCallbackId();
-  String getActionTs();
   String getToken();
-
   SlackTeam getTeam();
-  SlackChannel getChannel();
   SlackUserLite getUser();
+  Optional<SlackChannel> getChannel();
```

Comment:
Oof I see what you mean. Logistically it feels like this method should have lived on the discrete subclasses that actually have it (i.e. a `MessageAction` should still have a non-optional channel field). 

I'm really sad about breaking the API here, but I do recognize the reason is a mistake I made initially including this. @jhaber do you have any advice on the cleanest way to recover from an API level mistake like this? This field shouldn't have been on the base class, but now it's out in the wild.

Reply:
Chatted a bit with @stevegutz and @wsorenson about the implications of this change. This shouldn't have been done as it was, and it's not an API change but rather a build failure (and one that shouldn't be too hard to correct). In light of that, we're going to just roll ahead (since it will even be ok in many situations at present since folks don't extend this class, generally).

Reply:
Can you explain what you mean by "build failure"? A build failure where? This change can break (at runtime) any code that extends the base class. I don't think we need to worry about that, but I want to make sure this is documented in the PR.

Reply:
I didn't look closely at the change, but I suppose it brings into question if any of these options are available:

1) Only use explicit library versions in Maven.
2) Provide a second hierarchy of models and deprecate the original
3) Rename this class

Do any of these make sense?

Reply:
I'm in favour of (1), since that's kinda what versions are for.  2 & 3 are tricky because these classes are used for deserialisation of the messages received from slack.